xds/README.md

```diff
@@ -7,115 +7,153 @@ upstream endpoints and filter configuration from some external source, and then
+```
+
+> A proxy's pod must have a `quilkin.dev/role` annotation set to the value `proxy` in order
+  for the management server to detect the pod as a proxy and push updates to it.
+
+> Note that currently, the server can only discover resources within a single cluster.
+
+##### Admin server
+
+In addition the gRPC server, a http server (configurable via `--admin-port`is also started to serve administrative functionality.
+The following endpoints are provided:
+- `/ready`: Readiness probe that returns a 5xx if communication with the Kubernetes api is problematic.
+- `/live`: Liveness probe that always returns a 200 response.
+- `/metrics`: Exposes Prometheus metrics.
+
+##### Metrics
+
+The following metrics are exposed by the management server.
+
+- `quilkin_management_server_connected_proxies` (Gauge)
+
+   The number of proxies currently connected to the server.
+- `quilkin_management_server_discovery_requests_total{request_type}` (Counter)
+
+   The total number of xDS Discovery requests received across all proxies.
+   - `request_type` = `type.googleapis.com/envoy.config.cluster.v3.Cluster` | `type.googleapis.com/envoy.config.endpoint.v3.ClusterLoadAssignment` | `type.googleapis.com/envoy.config.listener.v3.Listener`
+     Type URL of the requested resource
+- `quilkin_management_server_discovery_responses_total` (Counter)
+
+   The total number of xDS Discovery responses sent back across all proxies in response to Discovery Requests.
+   Each Discovery response sent corresponds to a configuration update for some proxy.
+   - `request_type` = `type.googleapis.com/envoy.config.cluster.v3.Cluster` | `type.googleapis.com/envoy.config.endpoint.v3.ClusterLoadAssignment` | `type.googleapis.com/envoy.config.listener.v3.Listener`
+     Type URL of the requested resource
+- `quilkin_management_server_endpoints_total` (Gauge)
+
+   The number of active endpoints discovered by the server. The number of active endpoints
+   correlates with the size of the cluster configuration update sent to proxies.
+- `quilkin_management_server_snapshot_generation_errors_total` (Counter)
+
+   The total number of errors encountered while generating a configuration snapshot update for a proxy.
+- `quilkin_management_server_snapshots_generated_total` (Counter)
+
+   The total number of configuration snapshot generated across all proxies. A snapshot corresponds
+   to a point in time view of a proxy's configuration. However it does not necessarily correspond
+   to a proxy update - a proxy only gets the latest snapshot so it might miss intermediate
+   snapshots if it lags behind.
+- `quilkin_management_server_snapshots_cache_size` (Gauge)
+
+   The current number of snapshots in the in-memory snapshot cache. This corresponds 1-1 to
+   proxies that connect to the server. However the number may be slightly higher than the number
+   of connected proxies since snapshots for disconnected proxies are only periodically cleared
+   from the cache.
+
+##### File Server
```

Comment:
To match size and semantic level as the other section.

```suggestion
## Example: File Server Implementation
```

(Not sure if I'm sold on the heading content change, but something like this to make it clear on scan through that it is an example, vs the Agones one which is aimed at being used in production)